lib/internal/test_runner/utils.js

```diff
@@ -74,10 +81,63 @@ function convertStringToRegExp(str, name) {
   }
 }
 
+const kBuiltinDestinations = new SafeMap([
+  ['stdout', process.stdout],
+  ['stderr', process.stderr],
+]);
+
+const kBuiltinReporters = new SafeMap([
+  ['spec', specReporter],
+  ['dot', dotReporter],
+  ['tap', tapReporter],
+]);
+
+const kDefaultReporter = 'tap';
+const kDefaltDestination = 'stdout';
+
+function getReportersMap(reporters, destinations) {
+  const result = [];
+  reporters.forEach((name, i) => {
+    const destination = kBuiltinDestinations.get(destinations[i]) ?? createWriteStream(destinations[i]);
+    let reporter = kBuiltinReporters.get(name) ?? require(name);
```

Comment:
Are reporters specified in ESM supported? I think we'll need to support both ESM and CJS here.